### PR TITLE
Use a consistent build id between bazel/kubeadm pull jobs.

### PR DESCRIPTION
**Pull request by pipejakob**
Since the kubeadm e2e job uses the CI artifacts pushed by the bazel job, make sure we use a consistent id for their storage (the Kubernetes version isn't stable for pull jobs since each job pulls and applies the PR at different times, resulting in different commit SHAs).

The `PULL_NUMBER` isn't actually required for uniqueness of this location, but having it as part of the hierarchy makes it easy to find all bazel builds for a PR since this identifier is much friendlier than `PULL_REFS`, e.g.:

    $ gsutil ls gs://kubernetes-release-dev/bazel/44362/
    gs://kubernetes-release-dev/bazel/44362/master:ef2edcc94ef07fd782569e72b0da6255b96e3e76,44362:ebe32554395c85b42ee34d7f5901a211bbf80cf1/

### CODE REVIEW

**Comment by fejta** [comment]
/assign

I would like the two of us to work on a plan to migrate these jobs (and all the other kubeadm jobs) to a python scenario like the other PR jobs: https://github.com/kubernetes/test-infra/blob/master/jobs/config.json#L286

**Comment by pipejakob** [comment]
Sure, this particular job (the pull variant) is the only one that uses a shell script, and should be easy to convert. The CI and periodic variants of kubeadm e2e jobs are all scenario-based.

**Comment by fejta** [comment]
Any chance I can convince you to do this in this PR or would you prefer to defer to a later one?

**Comment by pipejakob** [comment]
Sure. I think the two changes are fundamentally isolated, though, so I'll keep them as separate commits.

**Comment by fejta** [comment]
/lgtm

I would really prefer to see these shell scripts die in a fiery ball of death

**Comment by pipejakob** [comment]
Okay, I've added a second commit to convert the pull job to the `kubernetes_e2e.py` scenario like the other kubeadm variants. Care to take another look?

**Comment by pipejakob** [comment]
(After I fix all of those new bazel failures, of course!)

File: scenarios/kubernetes_e2e.py [hunk 1]
```diff
@@ -313,28 +339,15 @@ def main(args):
     cluster = args.cluster or 'e2e-gce-%s-%s' % (
         os.environ['NODE_NAME'], os.getenv('EXECUTOR_NUMBER', 0))
 
-    if args.kubeadm:
+    if args.kubeadm is not None:
```
Comment (fejta) [inline scenarios/kubernetes_e2e.py]: `if not args.kubeadm`

https://google.github.io/styleguide/pyguide.html?showone=True/False_evaluations#True/False_evaluations

**Comment by fejta** [comment]
Fabulous, thanks so much for doing this!

/cc @krzyzacy

File: prow/config.yaml [hunk 2]
```diff
@@ -252,6 +254,8 @@ presubmits:
         - "--pull=$(PULL_REFS)"
         - "--upload=gs://kubernetes-jenkins/pr-logs"
         - "--git-cache=/root/.cache/git"
+        - "--timeout=75"
+        - "--json"
```
Comment (krzyzacy) [inline prow/config.yaml]: maybe now --json can be in the image

**Comment by krzyzacy** [comment]
/lgtm
no blocking, it looks like a better world :-)
please rebase

File: scenarios/kubernetes_e2e.py (hunk 1 above)
Comment (pipejakob) [inline scenarios/kubernetes_e2e.py]: Hunh, I thought I was having trouble with that syntax after changing this from a boolean, but it definitely works. I've reverted it back to the original "`if args.kubeadm:`".

File: prow/config.yaml (hunk 2 above)
Comment (pipejakob) [inline prow/config.yaml]: Oh yeah, good point. I'll add a personal TODO to follow this up with another PR.

**Pull request merged**